src/Mvc/Mvc.ViewFeatures/src/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -212,6 +212,7 @@ internal static void AddViewServices(IServiceCollection services)
             services.TryAddScoped<NavigationManager, HttpNavigationManager>();
             services.TryAddScoped<IJSRuntime, UnsupportedJavaScriptRuntime>();
             services.TryAddScoped<INavigationInterception, UnsupportedNavigationInterception>();
+            services.TryAddScoped<IComponentActivator, ComponentActivator>();
```

Comment:
Also can be removed.

Reply:
Couldn't the default activator be a singleton? It doesn't hold any state.